rust-error-handling-book/structure.md

```diff
@@ -0,0 +1,82 @@
+# Book
+
+## Introduction
+
+- General introduction (for inspiration, can look at these resources)
+  - [Rust CLI Book](https://rust-cli.github.io/book/index.html)
+  - [Cargo Book](https://doc.rust-lang.org/cargo/index.html)
+  - [The book](https://doc.rust-lang.org/book/ch00-00-introduction.html)
+- Purpose of the book
+  - To give an introduction to error handling in Rust
+  - To give practical solutions to error handling design
+    - Discussing different design decisions and tradeoffs
+  - Share best practices for error handling in different scenarios
+
+## Introduction to error handling
+
+- What is error handling
+  - Jane - [Error Handling is not all about errors](https://youtu.be/WzC7H83aDZc) (from minute 3:30).
+    - Defining errors
+    - Propagating errors and gathering context
+    - Reacting errors
+    - Discarding errors
+    - Reporting errors and gathered context
+- What is the purpose of errors?
+  - Communication
+    - All about expressing intent to the developer i.e this was supposed to happen but it failed and
+      as a result this other thing happened
+    - Preventing UB (but at the application level - i.e stopping invalid data or weird scenarios from
+      occurring because of mishandled errors)
+    - Giving enough context to failures to diagnose and debug rapidly
+- Anatomy of errors (captured in what is error handling but could be separated or reiterated)
+  - Creating
+  - Propagating
+  - Reporting
+  - Discarding
+  - Promoting (recoverable → irrecoverable)
+  - Reacting
+
+## Basic error handling in Rust
+
+- Could replace basic with another word
+
+- How Rust does error handling - do we need this since it's in The book? Possibly just to augment the
+  discussion from the book or just rehash it to make this book complete in its own right.
+  - Augmented material we could add:
+    - Some of the benefits of returning errors (e.g explicit, communicates intent) - discuss without
+      turning into a war
```

Comment:
I don't understand, what do you mean by "turning into a war"?

Reply:
Ahh going into a discussion between returning errors and exceptions. I think it would be better to just discuss returning errors and it's advantages without referencing why exceptions don't cater well for that need unless it's an objective statement